libs/langchain/src/agents/middlewareAgent/middleware/hitl.ts

```diff
@@ -1,33 +1,118 @@
  */
-export interface ToolApprovalRequest {
+export interface HumanInTheLoopConfig
+  extends Omit<ToolConfigSchema, "description"> {}
+
+/**
+ * Represents a request with a name and arguments.
+ */
+export interface ActionRequest {
+  /**
+   * The type or name of action being requested (e.g., "add_numbers")
+   */
   action: string;
+  /**
+   * Key-value pairs of arguments needed for the action (e.g., {"a": 1, "b": 2})
+   */
   args: Record<string, any>;
-  toolCallId: string;
+}
+
+/**
+ * Represents an interrupt triggered by the graph that requires human intervention.
+ *
+ * @param actionRequest - The specific action being requested from the human
+ * @param config - Configuration defining what response types are allowed
+ * @param description - Optional detailed description of what input is needed
+ *
+ * @example
+ * ```ts
+ * const hitlRequest: HumanInTheLoopRequest = {
+ *   actionRequest: { action: "Approve XYZ action", args: { ... } },
+ *   config: { allowAccept: true, allowEdit: true, allowRespond: true },
+ *   description: "Please review the command before execution"
+ * };
+ * response = interrupt([request])[0]
+ * ```
+ */
+export interface HumanInTheLoopRequest {
+  actionRequest: ActionRequest;
+  config: HumanInTheLoopConfig;
+  description?: string;
+}
+
+/**
+ * Response when a human approves the action.
+ */
+export interface AcceptPayload {
+  type: "accept";
+}
+
+/**
+ * Response when a human rejects the action.
+ */
+export interface ResponsePayload {
+  type: "response";
+  args?: string;
+}
+
+/**
+ * Response when a human edits the action.
+ */
+export interface EditPayload {
+  type: "edit";
+  args: ActionRequest;
+}
+
+export type HumanInTheLoopMiddlewareHumanResponse =
+  | AcceptPayload
+  | ResponsePayload
+  | EditPayload;
+
+/**
+ * Configuration for a tool requiring human in the loop.
+ */
+export interface ToolConfig extends HumanInTheLoopConfig {
+  /**
+   * The description attached to the request for human input
+   */
   description?: string;
 }
 
 const contextSchema = z
   .object({
-    toolConfigs: z
-      .record(
-        z.object({
-          requireApproval: z.boolean().optional(),
-          description: z.string().optional(),
-        })
-      )
-      .default({}),
-    messagePrefix: z.string().default("Tool execution requires approval"),
+    /**
+     * Mapping of tool name to allowed actions.
+     * If a tool doesn't have an entry, it's auto-approved by default.
+     *
+     * - `true` -> all actions allowed
+     * - `false` -> indicates that the tool is auto-approved.
+     * - `ToolConfig` -> indicates the specific actions allowed for this tool.
+     */
+    toolConfigs: z.record(z.union([z.boolean(), ToolConfigSchema])).default({}),
+    /**
+     * The prefix to use when constructing action requests.
+     * This is used to provide context about the tool call and the action being requested.
```

Comment:
Could we explain that it's only used for tools that do not a custom description. if the tools have a custom description (via ToolConfigSchema) the descriptionPrefix is not used for those tools